projects/arcadia-finance-v2/helper/index.js

```diff
@@ -0,0 +1,237 @@
+    pools: {
+      wethPool: "0x803ea69c7e87D1d6C86adeB40CB636cC0E6B98E2",
+      usdcPool: "0x3ec4a293Fb906DD2Cd440c20dECB250DeF141dF1",
+    },
+    nftPosManager: "0x03a520b32C04BF3bEEf7BEb72E919cf822Ed34f1",
+    univ3_factory: "0x33128a8fC17869897dcE68Ed026d694621f6FDfD",
+  },
+};
+
+const POOL_INIT_CODE_HASH =
+  "0xe34f199b19b2b4f47f68442619d555527d244f78a3297ea89325f843f87b8b54";
+const ZERO = JSBI.BigInt(0);
+const ONE = JSBI.BigInt(1);
+const Q32 = JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(32));
+const MaxUint256 = JSBI.subtract(
+  JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(256)),
+  JSBI.BigInt(1)
+);
+
+function computeAddress(factory, token0, token1, fee) {
+  const poolkey = Web3.utils.keccak256(
+    Web3.eth.abi.encodeParameters(
+      ["address", "address", "uint24"],
+      [token0, token1, fee]
+    )
+  );
+  const poolAddressFromSdk = getCreate2Address(
+    factory,
+    poolkey,
+    POOL_INIT_CODE_HASH
+  );
+  return Web3.utils.toChecksumAddress(poolAddressFromSdk);
+}
+
+function mulDiv(a, b, denominator) {
+  let product = JSBI.multiply(JSBI.BigInt(a), JSBI.BigInt(b));
+  return JSBI.divide(product, JSBI.BigInt(denominator));
+}
+
+function getAmountsFromLiquidity(
+  sqrtRatioX96,
+  tickLower,
+  tickUpper,
+  liquidity
+) {
+  const sqrtRatioAX96 = getSqrtRatioAtTick(tickLower);
+  const sqrtRatioBX96 = getSqrtRatioAtTick(tickUpper);
+  let amount0,
+    amount1 = 0;
+
+  if (JSBI.lessThanOrEqual(sqrtRatioX96, sqrtRatioAX96)) {
+    amount0 = getAmount0ForLiquidity(sqrtRatioAX96, sqrtRatioBX96, liquidity);
+  } else if (JSBI.lessThan(sqrtRatioX96, sqrtRatioBX96)) {
+    amount0 = getAmount0ForLiquidity(sqrtRatioX96, sqrtRatioBX96, liquidity);
+    amount1 = getAmount1ForLiquidity(sqrtRatioAX96, sqrtRatioX96, liquidity);
+  } else {
+    amount1 = getAmount1ForLiquidity(sqrtRatioAX96, sqrtRatioBX96, liquidity);
+  }
+
+  return { amount0, amount1 };
+}
+
+function getAmount0ForLiquidity(sqrtRatioAX96, sqrtRatioBX96, liquidity) {
+  if (JSBI.greaterThan(sqrtRatioAX96, sqrtRatioBX96)) {
+    [sqrtRatioAX96, sqrtRatioBX96] = [sqrtRatioBX96, sqrtRatioAX96];
+  }
+
+  const X96 = JSBI.BigInt(96);
+
+  let result = mulDiv(
+    JSBI.leftShift(liquidity, X96),
+    JSBI.subtract(sqrtRatioBX96, sqrtRatioAX96),
+    sqrtRatioBX96
+  );
+  result = JSBI.divide(result, sqrtRatioAX96);
+  return result;
+}
+
+function getAmount1ForLiquidity(sqrtRatioAX96, sqrtRatioBX96, liquidity) {
+  if (JSBI.greaterThan(sqrtRatioAX96, sqrtRatioBX96)) {
+    [sqrtRatioAX96, sqrtRatioBX96] = [sqrtRatioBX96, sqrtRatioAX96];
+  }
+  const denominator = JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(96));
+  let result = mulDiv(
+    liquidity,
+    JSBI.subtract(sqrtRatioBX96, sqrtRatioAX96),
+    denominator
+  );
+  return result;
+}
+
+function mulShift(val, mulBy) {
+  return JSBI.signedRightShift(
+    JSBI.multiply(val, JSBI.BigInt(mulBy)),
+    JSBI.BigInt(128)
+  );
+}
+
+function getSqrtRatioAtTick(tick) {
```

Comment:
I think defillama helpers for these, but when we create pr they will propose the changes